### PR TITLE
Hour input

### LINKED ISSUE #646: [FEATURE] Make hour Input version match the BI doc version
**Describe the solution you'd like**

It currently uses the native `time` input UI in the browser, but it wuld be better to match the BI version.

Extend the `Input` component (time version) based on the original Bootstrap Italia documentation:
https://italia.github.io/bootstrap-italia/docs/form/input-ora/

* [ ] Extend the `Input` component to handle time as in BI
* [ ] Add stories/examples for it

Notes:

* use BI classes to style the component.
* minimize the amount of extra CSS to style in this component ( possibly creating a set of reusable `react-select` styles objects reusable
* Probably build the component logic or add an external dependency to match it

### PR DESCRIPTION
**Pull request by NicolaFerracin**
Fixes #646 

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (if applicable).
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable).
- [x] I have added necessary documentation (if appropriate).

#### Short description of what this resolves:

Create a new `Input` component for displaying the selection of time (hh:mm).

#### Changes proposed in this Pull Request:

- add new props to Input (onBlur and onFocus callbacks + noWrapper prop to display the raw input without container)
- create `TimeInput` component with logic
- Tests

### CODE REVIEW

File: src/Input/Input.tsx [hunk 1]
```diff
@@ -89,29 +89,42 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    * @deprecated
    */
   static?: boolean;
+  /** Quando attivo rimuove il componente contenitore dell'Input. Utile per un controllo maggiore dello styling */
+  noWrapper?: boolean;
+  /** Funzione callback per quando il campo riceve il focus */
+  onFocus?: React.FocusEventHandler<HTMLInputElement>;
+  /** Funzione callback per quando il campo perde il focus */
+  onBlur?: React.FocusEventHandler<HTMLInputElement>;
```
Comment (dej611) [inline src/Input/Input.tsx:97]: These should be already inherited by `InputHTMLAttributes<HTMLInputElement>`, so no need to redefined them again.

File: src/Input/Input.tsx [hunk 2]
```diff
@@ -89,29 +89,42 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
...
 }
 
 type InputState = { isFocused: boolean; hidden: boolean; icon: boolean };
 
 export class Input extends React.Component<InputProps, InputState> {
+  public static defaultProps = {
+    onFocus: noop,
+    onBlur: noop
+  };
```
Comment (dej611) [inline src/Input/Input.tsx:106]: No need to define default values here.

**Review by dej611 — COMMENTED** [review]
🎉 Thanks for the contribution!

Left just a couple of minor comments, but the PR looks good 🚀

File: src/Input/Input.tsx (hunk 1 above)
Comment (NicolaFerracin) [reply src/Input/Input.tsx]: done!

File: src/Input/Input.tsx (hunk 2 above)
Comment (NicolaFerracin) [reply src/Input/Input.tsx]: right, we are already checking if they are defined, my bad

**Review by dej611 — APPROVED** [review]
Great! Thanks for the new component!

**Pull request merged**